packages/dds/tree/docs/roadmap.md

```diff
@@ -370,7 +370,19 @@ In this milestone, Shared Tree exposes an extension point that allows developers
 
 > Pending
 
-// TODO
+While [constraints](#constraints) and [schema](#schema-and-schema-enforcement) offer powerful ways to ensure an application's data model is never violated, they enable such guarantees by explicitly creating conflicts when their invariants are violated.
+These conflicts, if the application does not react to them, can be a form of data loss if the rejected change contained data a client cares about.
+Attempting to repair the outcome of a merge of conflicting edits “after the fact” is a challenging problem, one whose difficulty will grow non-linearly as the number and complexity of edits increases.
+
+An approach that places far less burden on application developers is to let them structure their edits in a manner that captures end users’ intent as precisely as possible.
+This model requires that edits are structured as _commands_ who's parameters are organized into tree references that the system understands and opaque parameters understood only by the command author.
+When a local edit is rebased against a conflicting edit by another user, the local edit can simply be rolled back, the conflicting edit applied first, and the command that produced the local edit then re-executed, as if the user had applied it under these new circumstances in the first place.
+The Shared Tree can use heuristics to adjust the tree reference parameters as necessary before the command is executed, meaning the handles passed to the command may be a best approximation of those originally supplied when the user first ran the command.
+Even if the command author does nothing else, many merge scenarios will yield better outcomes with this model than one that attempts to modify primitive operations applied to the tree.
+
+As an example, if User A adds a row to a table and User B concurrently adds a column, the table will be well-formed regardless of the eventual order of these edits, as would be the case if they were executed in either order on a single client.
+
+This milestone explores this command model, how it interplays with other features such as constraints, and how it handles edge-cases such as code availability and versioning.
```

Comment:
"explores" is quite vague - is there an actual deliverable? E.g. Commands will have been prototyped? Or fully designed and verified to work with the rest of the architecture? Or maybe just a write-up doc?

Reply:
Agreed, it's vague, but we don't yet know what the deliverable will be. I'll circle back to this when we've firmed that up.